presto-product-tests/src/main/java/com/facebook/presto/tests/jdbc/JdbcTests.java

```diff
@@ -167,15 +191,66 @@ public void shouldGetTableTypes()
+    public void testSqlEscapeFunctions()
+            throws SQLException
+    {
+        if (usingTeradataJdbcDriver(connection)) {
+            // These functions, which are defined in the ODBC standard, are implemented within
+            // the Simba JDBC and ODBC drivers.  The drivers translate them into equivalent Presto syntax.
+            // The translated SQL is executed by Presto.  These tests do not make use of edge-case values or null
+            // values because those code paths are covered by other (non-Simba specifc) tests.
+
+            assertThat(query("select {fn char(40)}")).containsExactly(row("("));
+            assertThat(query("select {fn convert('2016-10-10', SQL_DATE)}")).containsExactly(row(Date.valueOf("2016-10-10")));
+
+            // This translates to: SELECT cast('1234.567' as DECIMAL).
+            // When casting to DECIMAL without parameters, Presto rounds to the nearest integer value.
+            assertThat(query("select {fn convert('1234.567', SQL_DECIMAL)}")).containsExactly(row(new BigDecimal(1235)));
+
+            assertThat(query("select {fn convert('123456', SQL_INTEGER)}")).containsExactly(row(123456));
+            assertThat(query("select {fn convert('123abcd', SQL_VARBINARY)}")).containsExactly(row("123abcd".getBytes()));
+            assertThat(query("select {fn dayofmonth(date '2016-10-20')}")).containsExactly(row(20));
+            assertThat(query("select {fn dayofweek(date '2016-10-20')}")).containsExactly(row(5));
+            assertThat(query("select {fn dayofyear(date '2016-10-20')}")).containsExactly(row(294));
+            assertThat(query("select {fn ifnull({fn ifnull(null, null)}, '2')}")).containsExactly(row("2"));
+            assertThat(query("select {fn ifnull('abc', '2')}")).containsExactly(row("abc"));
+            assertThat(query("select {fn ifnull(null, '2')}")).containsExactly(row("2"));
+            assertThat(query("select {fn lcase('ABC def 123')}")).containsExactly(row("abc def 123"));
+            assertThat(query("select {fn left('abc def', 2)}")).containsExactly(row("ab"));
+            assertThat(query("select {fn locate('d', 'abc def')}")).containsExactly(row(5));
+            assertThat(query("select {fn log(5)}")).containsExactly(row(1.60943791243));
+            assertThat(query("select {fn right('abc def', 2)}")).containsExactly(row("ef"));
+            assertThat(query("select {fn substring('abc def', 2)}")).containsExactly(row("bc def"));
+            assertThat(query("select {fn substring('abc def', 2, 2)}")).containsExactly(row("bc"));
+            assertThat(query("select {fn timestampadd(SQL_TSI_DAY, 21, date '2001-01-01')}")).containsExactly(row(Date.valueOf("2001-01-22")));
+            assertThat(query("select {fn timestampdiff(SQL_TSI_DAY,date '2001-01-01',date '2002-01-01')}")).containsExactly(row(365));
+            assertThat(query("select {fn ucase('ABC def 123')}")).containsExactly(row("ABC DEF 123"));
+        }
```

Comment:
please display warning when wrong jdbc driver is used and that the test is skipped